MusicApp.Identity.Application/Services/Implementations/IdentityService.cs

```diff
@@ -0,0 +1,121 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using MusicApp.Identity.Application.Repositories;
+using MusicApp.Identity.Application.Services.Interfaces;
+using MusicApp.Identity.Application.Validators;
+using MusicApp.Identity.Application.DTOs;
+using MusicApp.Identity.Domain.Entities;
+using MusicApp.Identity.Domain.Exceptions;
+using FluentValidation.Results;
+
+namespace MusicApp.Identity.Application.Services.Implementations;
+
+public class IdentityService : IIdentityService
+{
+    private readonly HttpContext _httpContext;
+    private readonly IUserRepository _userRepository;
+    private readonly Role _artistRole;
+    private readonly IMapper _mapper;
+    private readonly UserRegisterDtoValidator _userRegisterDtoValidator;
+    private readonly UserLoginDtoValidator _userLoginDtoValidator;
+
+    public IdentityService(
+        IHttpContextAccessor httpContextAccessor,
+        IUserRepository userRepository,
+        IRoleRepository roleRepository,
+        IMapper mapper)
+    {
+        _userRepository = userRepository;
+        _mapper = mapper;
+
+        _httpContext = httpContextAccessor.HttpContext;
+        _artistRole = roleRepository.GetRoleByTitle("artist");
```

Comment:
explain it please

Reply:
I have an artist role in the application. When registering, the user chooses whether he will be an artist or not, and if he chooses to be an artist, I need to add artist role to him. To do this, I take it from the RoleRepository and add it to User.Roles (List<Role>).

Reply:
I need the RoleRepository only for this one action, so I dont save it in the field, but take the artist role from it in the constructor.

Reply:
it's a little strange to get role in constructor. If you need this in scope of you method, please, get this role here and then do everything you want with it)